packages/components/src/components/ScalarSidebar/ScalarSidebarNestedItems.vue

```diff
@@ -0,0 +1,149 @@
+<script lang="ts">
+/**
+ * Scalar Sidebar Nested Items component
+ *
+ * A provides list of items thats presented over the parent list
+ * Needs to be nested inside a ScalarSidebarItems component
+ *
+ * @example
+ * <ScalarSidebarNestedItems v-model="open">
+ *   <!-- Item text -->
+ *   <template #items>
+ *     <ScalarSidebarItem>...</ScalarSidebarItem>
+ *     <ScalarSidebarItem>...</ScalarSidebarItem>
+ *     <ScalarSidebarItem>...</ScalarSidebarItem>
+ *   </template>
+ * </ScalarSidebarNestedItems>
+ */
+export default {}
+</script>
+<script setup lang="ts">
+import {
+  ScalarIconArrowRight,
+  ScalarIconCaretLeft,
+  ScalarIconListDashes,
+} from '@scalar/icons'
+import { Transition, nextTick, ref, useTemplateRef, watch } from 'vue'
+
+import { ScalarIconLegacyAdapter } from '../ScalarIcon'
+import ScalarSidebarButton from './ScalarSidebarButton.vue'
+import ScalarSidebarItems from './ScalarSidebarItems.vue'
+import ScalarSidebarSpacer from './ScalarSidebarSpacer.vue'
+import { findScrollContainer } from './findScrollContainer'
+import type { ScalarSidebarItemProps } from './types'
+import { useSidebarGroups } from './useSidebarGroups'
+import { useSidebarNestedItem } from './useSidebarNestedItems'
+
+const { icon = ScalarIconListDashes } = defineProps<ScalarSidebarItemProps>()
+
+const open = defineModel<boolean>({ default: false })
+useSidebarNestedItem(open)
+
+defineSlots<{
+  /** The text content of the button */
+  'default'?: () => any
+  /** Override the entire button */
+  'button'?: () => any
+  /** Override the icon */
+  'icon'?: () => any
+  /** Override the aside slot */
+  'aside'?: () => any
+  /** Override the back button */
+  'back'?: () => any
+  /** Override the back button label */
+  'back-label'?: () => any
+  /** The list of sidebar subitems */
+  'items'?: () => any
```

Comment:
nit: not that it matters at all here, but its a good habit to use `unknown` over `any`